cmd/mistryd/metrics_recorder.go

```diff
@@ -0,0 +1,124 @@
+package main
+
+import (
+	"fmt"
+	"time"
+	"io/ioutil"
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var (
+	MetricsRecorderInstance *MetricsRecorder
+	MetricsRecorderOnce sync.Once
+)
+
+// Singleton access
+func GetMetricsRecorder() *MetricsRecorder {
```

Comment:
Instead of this function, we can just use a package-level global variable:

```go
var (
  MetricsRecorder = NewMetricsRecorder()
)

// ...
```

**Update**: Even better would be to make this just a member of the `Server` struct and initialize it inside `NewServer()`.

Reply:
This is a singleton implementation, that is why there is a `Get` function here. The singleton approach is an alternative to having the recorder as part of the server. 

I have no problem of instantiating the recorder in the server. The metrics are only related to the server anyway.

Reply:
I know. My point is that we can avoid the singleton since by definition we'll access this instance only via the server anyway.